header/sync/sync.go

```diff
@@ -76,10 +71,9 @@ func (s *Syncer) Start(context.Context) error {
 }
 
 // Stop stops Syncer.
-func (s *Syncer) Stop(context.Context) error {
+func (s *Syncer) Stop(ctx context.Context) error {
 	s.cancel()
-	s.cancel = nil
```

Comment:
why?

Reply:
Syncer can be stopped twice when:
1) BEFP is received;
2) node is stopping;

To avoid any potential races we agreed to remove ` if s.cancel != nil {`. Now cancel can be invoked twice but it's not a problem.

Reply:
ack